lib/utils/import/googlekeep/index.ts

```diff
@@ -0,0 +1,101 @@
+import { EventEmitter } from 'events';
+import { endsWith, isEmpty, get, has } from 'lodash';
+import JSZip from 'jszip';
+
+import CoreImporter from '../';
+
+let fs = null;
+const isElectron = has(window, 'process.type');
+if (isElectron) {
+  fs = __non_webpack_require__('fs'); // eslint-disable-line no-undef
+}
+
+class GoogleKeepImporter extends EventEmitter {
+  constructor({ noteBucket, tagBucket, options }) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+    this.options = options;
+  }
+
+  importNotes = filesArray => {
+    if (isEmpty(filesArray)) {
+      this.emit('status', 'error', 'No file to import.');
+      return;
+    }
+
+    const coreImporter = new CoreImporter({
+      noteBucket: this.noteBucket,
+      tagBucket: this.tagBucket,
+    });
+
+    let importedNoteCount = 0;
+
+    const importJsonString = jsonString => {
+      const importedNote = JSON.parse(jsonString);
```

Comment:
this could fail if the export file is corrupted or simply not what we expect.
could we emit an error here or should we skip any files we can't parse?

if all we wanted to do was skip them then we don't need to change the code since the `Exception` will be swallowed by the promise, but if we do that it would be ideal to indicate in a comment that we expect this to happen and that that's our expected resolution.

Reply:
I think just skipping failing notes is the best solution, emitting a `warning` status will display the error message only until the next update so it isn't much use for the user anyway. I'll add a comment mentioning that.

Ideally a warning for each failing note could show up in addition to the "complete" message but I imagine this would require more work.

Reply:
yeah skipping is fine for now.